backend/src/ee/services/permission/permission-service.ts

```diff
@@ -361,7 +376,14 @@ export const permissionServiceFactory = ({
     const scopes = ServiceTokenScopes.parse(serviceToken.scopes || []);
     return {
       permission: buildServiceTokenProjectPermission(scopes, serviceToken.permissions),
-      membership: undefined
+      membership: undefined,
+      ForbidOnInvalidProjectType: (productType: ProjectType) => {
```

Comment:
We know for service tokens, it is only used for secrets project so maybe we don't need this here?

Reply:
move to next pr